### DISCUSSION TITLE
nwg-shell Hyprland compatibility

### DISCUSSION THREAD

**Discussion by nwg-piotr** [nwg-piotr/nwg-shell#91] [General]
## Most recent answers to all the "does it work on Hyprland" questions:

I first installed Hyprland on March 20th. I took me an evening or two to explore the compositor and read the docs. Then I decided to try and make some or all the nwg-shell components Hyprland-compatible. Here they are in the order I planned the work:

| Component  | TODO                           | Progress |
| ------------------------ | -------------------------------- | ------------- |
| nwg-drawer         | - | ready OOTB |
| nwg-menu           | - | ready OOTB |
| nwg-look              | - | ready OOTB |
| utilities                 | - | ready OOTB |
| nwg-panel           |add HyprlandTaskbar   | ready since v0.8.2                  |
| Azote                     |minor adjustments        | ready since v1.11.0              |
| nwg-displays        | support Hyprland IP C | ready since v0.3.0 |
| nwg-dock-hyprland              | new app | ready        |
| nwg-panel           |add HyprlandWorkspaces   | ready since v0.8.5           |
| nwg-shell-config  | write a separate app    | under consideration  |
| installer  | ?    | under consideration  |

**Comment by kothavade** [comment]
Perhaps, in the interim, could you make it possible to build/run nwg-shell-config without anything sway specific? That way I can use the rest of its functionality until a separate app is built.

**Comment by nwg-piotr** [comment]
I'm hesitating between providing specific features depending on the environment, or writing a separate application for Hyprland, and I don't want to do unnecessary work that will take time. This topic will have to wait until nwg-dock is ready. Personally, I don't use it much, but I've had quite a few inquiries about a version that works on Hyprland.

Also, I need to know what's going to work well as the night light. Neither wlsunset, nor gammastep behave well on Hyprland & my hardware. Actually it's one of many issues to resolve.

**Reply by XxRiKxX** [reply]
Hyprland is getting even more popular than sway. So it might be a good idea to write something specific to hyprland. Love your work btw :)

**Reply by nwg-piotr** [reply]
Thank you!

One new thing was inevitable: a new dock. If it comes to the panel, Hyprland-specific modules will do the job. HyprlandTaskbar is ready, HyprlandWorkspaces in progress at the moment. The config utility will most probably require a new app.

**Reply by XxRiKxX** [reply]
Best of luck mate! Wish I could have helped in any way, but I am total noob. Using nwg shell made sway a hell lot easier for me.

**Reply by nwg-piotr** [reply]
Good to know, thanks. That's what it was all about.
